<!--

Programmer: Victoria Albanese
Filename: multiple_sensors_pcl.launch

Purpose: Launch multiple Intel RealSense 435i sensors
so that the pointclouds they generate can be visualized

-->

<launch>

	<!-- This launches each of the cameras with the correct parameters -->

	<include file="$(find realsense2_camera)/launch/rs_rgbd.launch">
		<arg name="camera" value="cam_1" />
		<arg name="serial_no" value="843112074058" />
	</include>

	<include file="$(find realsense2_camera)/launch/rs_rgbd.launch">
		<arg name="camera" value="cam_2" />
		<arg name="serial_no" value="843112073834" />
	</include>

	<include file="$(find realsense2_camera)/launch/rs_rgbd.launch">
		<arg name="camera" value="cam_3" />
		<arg name="serial_no" value="843112073646" />
	</include>


	<!-- This transforms each resulting pointcloud into a global reference frame -->

    <node name="camera_1_tf" pkg="tf" type="static_transform_publisher"
        args="0 0 2 0 0 0 /world /cam_1_link 100"/>

	
    <node name="camera_2_tf" pkg="tf" type="static_transform_publisher"
        args="0 0 0 0 0 0 /world /cam_2_link 100"/>

    <node name="camera_3_tf" pkg="tf" type="static_transform_publisher"
        args="0 0 -2 0 0 0 /world /cam_3_link 100"/>
	

	<!-- This launches RViz all setup to visualize them -->
	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find synchronize_pointclouds)/rviz/pointclouds.rviz" required="true" />

</launch>

